### PR TITLE
BUILD-2101 Use Vault

### PR DESCRIPTION
**Pull request by elrzn**
- [x] `.cirrus.yml`
- [x] `.github/workflows/release.yml`

### CODE REVIEW

**Comment by elrzn** [comment]
Error seems to be dev related.

**Comment by elrzn** [comment]
Hi @SonarSource/languages-team-cfamily can you maybe take a look? I believe the build error is not caused by the PR changes.

https://cirrus-ci.com/task/5534953210380288

**Review by Godin — CHANGES_REQUESTED** [review]
`.github/workflows/release.yml` should also be updated.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -9,35 +9,11 @@ env:
...
-          echo "${{ steps.lt_release.outputs.releasability }}"
-          echo "${{ steps.lt_release.outputs.release }}"
+  release:
+    permissions:
+      id-token: write
+      contents: write
+    uses: SonarSource/gh-action_release/.github/workflows/main.yaml@5.0.1
+    with:
+      publishToBinaries: true
+      mavenCentralSync: true
```
Comment (elrzn) [inline .github/workflows/release.yml:19]: Note: No Slack notifications as I don't know which channel to use.

File: .cirrus.yml [hunk 2]
```diff
@@ -2,20 +2,20 @@ gcp_credentials: ENCRYPTED[!149d4005ecdba4cdd78bb5ba22756ebb98bf8e3367ee2e9ab08c
...
-  ARTIFACTORY_PRIVATE_USERNAME: repox-private-reader-lt-1a7e1f
-  ARTIFACTORY_PRIVATE_PASSWORD: ENCRYPTED[!4890acae4038fb09d3921b1126aad6af4c0bc3984e603cf1639766e44cc987c3f2b7d529b5420b2e3eca0f354b938bf8!]
-  ARTIFACTORY_DEPLOY_USERNAME: repox-qa-deployer-lt-1a7e1f
-  ARTIFACTORY_DEPLOY_PASSWORD: ENCRYPTED[!91fd8560ac00c4661c3161af1bd6e74ed8de8799e9d6ddc2f07bbcf154703adfb54d197ec2286c25f481fca7aba18c76!]
+  CIRRUS_VAULT_AUTH_PATH: jwt-cirrusci
+  CIRRUS_VAULT_ROLE: cirrusci-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}
+  CIRRUS_VAULT_URL: https://vault.sonar.build:8200
+  ARTIFACTORY_URL: VAULT[development/kv/data/repox data.url]
+  ARTIFACTORY_PRIVATE_USERNAME: vault-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader
+  ARTIFACTORY_PRIVATE_PASSWORD: VAULT[development/artifactory/token/SonarSource-sslr-private-reader access_token]
```
Comment (Godin) [inline .cirrus.yml:10]:
```suggestion
  ARTIFACTORY_PRIVATE_PASSWORD: VAULT[development/artifactory/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-private-reader access_token]
```

File: .cirrus.yml [hunk 3]
```diff
@@ -2,20 +2,20 @@ gcp_credentials: ENCRYPTED[!149d4005ecdba4cdd78bb5ba22756ebb98bf8e3367ee2e9ab08c
...
+  ARTIFACTORY_DEPLOY_USERNAME: vault-${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-qa-deployer
+  ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/SonarSource-sslr-qa-deployer access_token]
```
Comment (Godin) [inline .cirrus.yml:12]:
```suggestion
  ARTIFACTORY_DEPLOY_PASSWORD: VAULT[development/artifactory/token/SonarSource-sonar-lits-qa-deployer access_token]
```

File: .cirrus.yml [hunk 4]
```diff
@@ -58,8 +58,9 @@ promote_task:
     memory: 500M
   env:
     #promotion cloud function
-    GCF_ACCESS_TOKEN: ENCRYPTED[!1fb91961a5c01e06e38834e55755231d649dc62eca354593105af9f9d643d701ae4539ab6a8021278b8d9348ae2ce8be!]
-    PROMOTE_URL: ENCRYPTED[!e22ed2e34a8f7a1aea5cff653585429bbd3d5151e7201022140218f9c5d620069ec2388f14f83971e3fd726215bc0f5e!]   
+    GCF_ACCESS_TOKEN: VAULT[development/kv/data/promote data.token]
+    PROMOTE_URL: VAULT[development/kv/data/promote data.url]
+    GITHUB_TOKEN: VAULT[development/github/token/SonarSource-sslr-promotion token]
```
Comment (Godin) [inline .cirrus.yml:63]:
```suggestion
    GITHUB_TOKEN: VAULT[development/github/token/${CIRRUS_REPO_OWNER}-${CIRRUS_REPO_NAME}-promotion token]
```

**Review by Godin — CHANGES_REQUESTED** [review]

**Review by Godin — APPROVED** [review]

**Pull request merged**